### DISCUSSION TITLE
Ottenere sensore temperatura con media di altri sensori

### DISCUSSION THREAD

**Discussion by Borghese** [indomus/forum#2851] [GENERALE - Discussioni generiche]
ciao a tutti
in ogni stanza della mia casa ho un sensore di temperatura/umidità lumi aqara zigbee, integrati in home assistant tramite ZHA. 
Volevo crearmi due sensori, uno che mi faccia la media delle temperatura, e uno per la media delle umidità, per inserirlo in una card nell'home page della dashboard (e poi linkato a una pagina con il dettaglio dei valori per stanza)
C'è un modo per creare un sensore facendo la media ? 

grazie a tutti

**Comment by snakuzzo** [comment]
Puoi usare i template...

https://indomus.it/formazione/esercizi-di-domotica-su-home-assistant-1-creare-dei-sensori-template/

Basta poi adeguare con il calcolo che vuoi fare tu

**Comment by Borghese** [comment]
grazie pèr la guida snakuzzo.
La sto seguendo ma già nel modello di stringa non ottengo il risultato sperato (mi restituisce zero anzichè la temperatura rilevata dal sensore) dove sto sbagliando ? 

```python
{{ state_attr("Studio Temperatura","sensor.studio_temperatura")|float(default=0)}}
```

**Reply by snakuzzo** [reply]
Allora ... o non hai letto bene o non hai capito bene. Quello che hai scritto è completamente sbagliato.
`"Studio Temperatura"` probabilmente è il friendly name dell'entità, non il nome dell'entità. La funzione `state_attr()` non si usa come hai fatto tu....va passato prima il nome dell'entità e poi il nome dell'attributo. Inoltre, probabilmente non dovrai usare questa funziona, ma `states()`

Ora... io non so che cavolo di entità/device hai ma, a seconda di come viene esposto quel valore, lo si estrae...

Così...se il valore è lo `state` dell'entità...

```python
{{ states('sensor.sensore_temperatura') }}
```

oppure, se è l'attributo di un'entità...

```python
{{ state_attr('sensor.sensore_temperatura','temperatura') }}
```

...poi aggiungi il cast a float con un valore di default

**Reply by Borghese** [reply]
si hai ragione, ci sono arrivato dopo ma tu sei stato piu veloce di me a risp. ;)

Così funziona : 
```python
{{ states('sensor.studio_temperatura')|float(default=0)}}
``` 

Però non mi è ancora chiaro come creare la media di "n" di questi sensori... nella guida non ho visto un esercizio di questo tipo o sbaglio ?

**Reply by snakuzzo** [reply]
Premesso che puoi semplificarti la vita usando delle integrazioni apposite (es. [min/max](https://www.home-assistant.io/integrations/min_max/)), ecco come puoi fare...

```python
"Temperatura soggiorno"
{{ states('sensor.soggiorno_temperature') | float }}

"Temperatura ingresso"
{{ states('sensor.ingresso_temperature') | float }}

"Media"
{{ ( (states('sensor.soggiorno_temperature') | float + states('sensor.ingresso_temperature') | float) / 2) | round(1) }}
```

Dimostrazione...

![image](https://github.com/indomus/forum/assets/7205371/1c1d3d98-0dae-479a-a38a-a9d8b8329b69)

**Reply by Borghese** [reply]
Grazie Snakuzzo!! ho provato l'integrazione e in effetti funziona alla grande per il mio scopo.
Devo solo capire se più corretto utilizzare la media aritmetica o la mediana, per avere la temperatura media di casa data dai 7 sensori, ma per il resto è super!

**Comment by KC-inDomus** [comment]
<img width="678" alt="Screenshot 2024-01-22 at 08 11 11" src="https://github.com/indomus/forum/assets/52242036/81391fbd-0050-41bf-bc8a-9a4ee7a91a90">
<img width="641" alt="Screenshot 2024-01-22 at 08 12 17" src="https://github.com/indomus/forum/assets/52242036/e9d2df7c-f89f-4b94-807b-478e39012736">


Fai prima a usare un helper, sebbene non ti faccia affatto male imparare a definirti dei sensori template.

**Reply by Borghese** [reply]
grazie KC-inDomus ho installato proprio quell'integrazione e sto usando l'aiutante per ricavare il valore medio dei sensori temp/umidità
